.github/workflows/ci.yml

```
@@ -632,6 +632,61 @@ jobs:
         run: env
         if: ${{ always() }}
 
+  macos-monterey:
+    name: "x86_64: macOS Monterey, ctime_tests"
+    # See: https://github.com/actions/runner-images#available-images.
+    runs-on: macos-12
+
+    env:
+      CC: 'clang'
+      HOMEBREW_NO_AUTO_UPDATE: 1
+      HOMEBREW_NO_INSTALL_CLEANUP: 1
+
+    strategy:
+      fail-fast: false
+      matrix:
+        env_vars:
+          - { WIDEMUL: 'int64',  RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes' }
+          - { WIDEMUL: 'int64',  RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes', CC: 'gcc' }
+          - { WIDEMUL: 'int128_struct', ECMULTGENPRECISION: 2, ECMULTWINDOW: 4 }
+          - { WIDEMUL: 'int128',                  ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes' }
+          - { WIDEMUL: 'int128', RECOVERY: 'yes',              SCHNORRSIG: 'yes' }
+          - { WIDEMUL: 'int128', RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes', CC: 'gcc' }
```

I suggest this merged matrix, which also includes real Valgrind jobs (see the main comment in my review for details): 
```suggestion
        env_vars:
          - { WIDEMUL: 'int64',  RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes' }
          - { WIDEMUL: 'int128_struct', ECMULTGENPRECISION: 2, ECMULTWINDOW: 4 }
          - { WIDEMUL: 'int128',                  ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes' }
          - { WIDEMUL: 'int128', RECOVERY: 'yes' }
          - { WIDEMUL: 'int128', RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes' }
          - { WIDEMUL: 'int128', RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes', CC: 'gcc' }
          - { WIDEMUL: 'int128', RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes',            WRAPPER_CMD: 'valgrind --exit-code=42' }
          - { WIDEMUL: 'int128', RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes', CC: 'gcc', WRAPPER_CMD: 'valgrind --exit-code=42' }
          - { WIDEMUL: 'int128', RECOVERY: 'yes', ECDH: 'yes', SCHNORRSIG: 'yes', ELLSWIFT: 'yes', CPPFLAGS: '-DVERIFY' }
          - BUILD: 'distcheck'
```

---

By the way, I don't like this style of having the same job with and without valgrind. Perhaps we should reintroduce a `RUN_ON_VALGRIND` variable (removed in fcfcb97e74b55a107290d44c81c049d6168e954f), but that should be a separate PR, and I don't think it should hold up this PR here.

---

> I suggest this merged matrix, which also includes real Valgrind jobs (see the main comment in my review for details):

Thanks! Implemented in https://github.com/bitcoin-core/secp256k1/pull/1412#issuecomment-1695937918.

---

Ah, I think we should set `SECP256K1_TEST_ITERS=2` like for the other Valgrind jobs....

Perhaps it's really cleaner to have a `RUN_ON_VALGRIND`, and the script then reduces the number of iterations automatically by a factor of 32 or 16...  

> but that should be a separate PR, and I don't think it should hold up this PR here.

I stand by this, so I'd say do whatever you prefer.

---

> Ah, I think we should set `SECP256K1_TEST_ITERS=2` like for the other Valgrind jobs....

Sure. Done.